libs/common/src/types/mongo-db-models/deals.ts

```diff
@@ -2,10 +2,12 @@ import { ObjectId } from 'mongodb';
 import { DEAL_TYPE } from '../../constants';
 import { AnyObject } from '../any-object';
 import { DealSubmissionType } from '..';
+import { Activity } from '../portal';
 
 type BaseDeal = AnyObject & {
   _id: ObjectId;
   submissionType: DealSubmissionType | null;
+  portalActivities: Array<Activity>;
```

Comment:
Should this be on the GefDeal? also you use `Activity[]`

Reply:
I tried to have this in `GefDeal` only, but TS do not like it. I'm not sure why, but with this change, in other places, It thinks `portalActivities` is not a recognised type.

Reply:
It looks like its a genuine type error in `DealCancellationService.submitDealCancellation`

In the following line, `portalActivities` is being passed in even when the deal might be BSS/EWCS.
```
    await PortalDealService.addGefDealCancelledActivity({ dealId, dealType, portalActivities, author, auditDetails });
```

To get around this, you could do something like
- add `portalActivities: undefined` to BssEwcsDeal explitly & add something like `portalActivities ?? []` to the DealCancellationService
or
- Check if the deal is `GEF` in `DealCancellationService.submitDealCancellation` & only call `addGefDealCancelledActivity` if it is. You can then use `as Activity[]` truthfully
or
- Pass in the entire deal to `addGefDealCancelledActivity` as type `Deal` & then within `addGefDealCancelledActivity` check if its GEF & the type should automatically update free of charge.

The latter is probably the easiest & maybe nicest - we're passing in `dealId`, `dealType` and `activities` anyway so it maybe is nicer to group them into the original deal object

Reply:
I'd be quite keen to fix this since otherwise our types arent accurate & at a glance it would seem that `portalActivities` do exist on BSS/EWCS deals